core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -285,8 +287,20 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
         sqp.setDefaultOperator(defaultOperator.toBooleanClauseOccur());
 
         Query query = sqp.parse(queryText);
-        if (minimumShouldMatch != null && query instanceof BooleanQuery) {
-            query = Queries.applyMinimumShouldMatch((BooleanQuery) query, minimumShouldMatch);
+        if (query instanceof BooleanQuery) {
+            BooleanQuery booleanQuery = (BooleanQuery) query;
+            if (booleanQuery.clauses().size() > 1
+                    && ((booleanQuery.clauses().get(0)).getQuery() instanceof BooleanQuery) == false) {
```

Comment:
Also, this doesn't really seem to make sense to me, since it could potentially do different things depending on the order of the clauses in the query (if it's a bool holding clauses like [TermQuery BooleanQuery] versus a bool holding [BooleanQuery TermQuery]). Am I misunderstanding something?